src/frequenz/sdk/actor/_power_managing/_bounds.py

```diff
@@ -0,0 +1,143 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Utilities for checking and clamping bounds and power values to exclusion bounds."""
+
+from ...timeseries import Bounds, Power
+
+
+def check_exclusion_bounds_overlap(
+    lower_bound: Power,
+    upper_bound: Power,
+    exclusion_bounds: Bounds[Power] | None,
+) -> tuple[bool, bool]:
+    """Check if the given bounds overlap with the given exclusion bounds.
+
+    Example:
+
+                       lower                        upper
+                          .----- exclusion zone -----.
+        -----|✓✓✓✓✓✓✓✓✓✓✓✓|xxxxxxxxxxxxxxx|----------|----
+             `-- usable --'-- exclusion --´
+             |                 overlap    |
+             |                            |
+           lower                        upper
+           bound                        bound
+                              (inside the exclusion zone)
```

Comment:
This will not be rendered for now as it is private, but if at some point we make it public, it will render incorrectly, so I suggest:

~~~suggestion
        ```
                       lower                        upper
                          .----- exclusion zone -----.
        -----|✓✓✓✓✓✓✓✓✓✓✓✓|xxxxxxxxxxxxxxx|----------|----
             `-- usable --'-- exclusion --´
             |                 overlap    |
             |                            |
           lower                        upper
           bound                        bound
                              (inside the exclusion zone)
        ```
~~~